packages/graph/hash_graph/lib/graph/src/shared/identifier/knowledge.rs

```diff
@@ -134,18 +135,42 @@ impl EntityVersion {
     }
 }
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, ToSql)]
+#[postgres(transparent)]
+#[repr(transparent)]
+pub struct EntityRecordId(i64);
+
+impl EntityRecordId {
+    #[must_use]
+    pub const fn new(id: i64) -> Self {
+        Self(id)
+    }
+
+    #[must_use]
+    pub const fn as_i64(&self) -> i64 {
+        self.0
+    }
+}
+
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Serialize, ToSchema)]
 #[serde(rename_all = "camelCase")]
 pub struct EntityEditionId {
     base_id: EntityId,
+    #[serde(skip)]
```

Comment:
I wouldn't like to expose it as it's likely, that we will have breaking changes in this struct anyway. If we are happy with the struct and want to use it in the backend we can still remove the skipping. 
I'd defer it until [Adjust structural queries for temporal versioning](https://app.asana.com/0/0/1203491211535116/f) _(internal)_.

Reply:
I agree that we will have breaking changes in the struct, but what I mean is that we can incrementally migrate.

- Let's say we _add_ `record_id`
- Next we update the workspace to stop using `version` and start using `record_id`
- Next you remove `version`

It allows us to do it in three steps, where 3 is technically a breaking change but wouldn't require updating the workspace because we'd already moved off it.

Basically a deprecation warning.

Reply:
But I don't feel strongly about this

Reply:
In order to use it, we also need to adjust the query token parameter and have to expose `"recordId"` as a valid field on entity queries, otherwise, it couldn't be used.
I'm fine by doing this, though, but let's keep that change out of this PR, so we have more clear bounds.